src/sp/transport/mqtt/broker_tcp.c

```diff
@@ -707,10 +707,17 @@ tcptran_pipe_recv_cb(void *arg)
 	// as application message callback of users
 	nni_aio_list_remove(aio);
 	msg      = p->rxmsg;
-	p->rxmsg = NULL;
 	n        = nni_msg_len(msg);
 	type     = p->rxlen[0] & 0xf0;
 
+	if (len <= 0 &&
+	    (type == CMD_SUBSCRIBE || type == CMD_PUBLISH ||
+	        CMD_UNSUBSCRIBE)) {
```

Comment:
But for packets with other types. Such as puback etc. The length also needs to be limited

Reply:
consider mqtt v5